### ISSUE TITLE
Please add docs/tutorial

### ISSUE THREAD

**Issue by suciokhan** [TokTok/py-toxcore-c#78]
I am super excited about this, I think it's exactly what I've been looking for. As a Python coder, I want to be able to script sending/receiving messages over a secure messaging application and have it trigger actions. I have found NO secure messaging applications with Python bindings. This has awesome potential; please flesh it out.

**Comment by iphydf** [comment]
Definitely we will have better documentation once all the functions are implemented. Currently this is a bit on hold because of lack of time (we're spending all our time on new DHT based group chats right now). Contributions towards exposing more toxcore functionality in the python api are more than welcome. Otherwise, I'll get back to it in a month or so after NGC (new group chats) are merged.

**Comment by emdee-is** [comment]
There needs to be documentation in the README of how much of the API is covered, and what's been left out. No point trying to use this if there is no coverage on the API calls you want to use. And you can't tell from the test coverage either.

@iphydf told me in a chat that there is a maintainer for this at the moment. You might want to try https://git.plastiras.org/emdee/toxygen_wrapper/ which is a CTYPES based wrapper, with very broad API coverage.

PS: NGC finally merged.

**Comment by emdee-is** [comment]
Still no documentation - the README **should** say that this is only a partial implementation and is not useable for anything serious.

Personally I think it was a mistake to "rewrite from scratch" in Cython when there was a good ctypes based set of bindings with full API coverage and good tests.

**Comment by iphydf** [comment]
The rewrite is blocked on some changes in toxcore. Also the old implementation wasn't ctypes. It was regular C bindings.

**Comment by emdee-is** [comment]
@iphydf I wasn't referring to the old code which was very fragile. At the time of that code there was also a ctypes wrapper, which has been updated and
has an integration testsuite: toxygen_wrapper - ctypes wrapping of libtoxcore into Python https://git.plastiras.org/emdee/toxygen_wrapper

To me, by rewriting from scratch in Cython and stopping has stopped all Tox development in Python for 2 years. And Cython has no advantages over ctypes for a fairly stable API like Tox.

**Comment by iphydf** [comment]
You're saying that rewriting the old hand written C bindings in this repository into cython has stopped all python development in tox. Can you explain the causality there?

**Comment by emdee-is** [comment]
Just a matter of personal opinion of course, but this repo is the focal point for Python for Tox. I tried the old C bindings back then and I think I got them to work, but replacing them with incomplete Cython bindings meant there have been no complete supported Python bindings for the project for 2 years.

IMHO, it would have been better to adopt into the main TokTok project the
ctypes bindings from https://github.com/toxygen-project/toxygen/tree/next_gen/toxygen/wrapper and maintained them so that the project had _visible_ support for Python. That old code supported NGC groups and only needed a little fixing for API changes. Maybe have kept the toxygen project alive.

I've added a fair amount of error reporting, and it's stable because the Tox API is stable, and ctypes is efficient. Plus I added an (incomplete) integration test suite.

**Comment by emdee-is** [comment]
See also the suggestions in https://github.com/TokTok/c-toxcore/issues/2702
